wallet/notifications.go

```diff
@@ -198,6 +198,57 @@ func makeTxSummary(dbtx walletdb.ReadTx, w *Wallet, details *udb.TxDetails) Tran
 	}
 }
 
+func makeTicketSummary(dbtx walletdb.ReadTx, w *Wallet, details *udb.TicketDetails) *TicketSummary {
+	var ticketStatus = TicketStatusLive
+
+	ticketTransactionDetails := makeTxSummary(dbtx, w, details.Ticket)
+	if details.Spender != nil {
+		spenderTransactionDetails := makeTxSummary(dbtx, w, details.Spender)
+		if details.Spender.TxType == stake.TxTypeSSGen {
+			ticketStatus = TicketStatusVoted
+		} else if details.Spender.TxType == stake.TxTypeSSRtx {
+			ticketStatus = TicketStatusRevoked
+		}
+		return &TicketSummary{
+			Ticket:  &ticketTransactionDetails,
+			Spender: &spenderTransactionDetails,
+			Status:  ticketStatus,
+		}
+	}
+
+	if details.Ticket.Height() == int32(-1) {
+		ticketStatus = TicketStatusUnmined
+	} else {
+		txmgrNs := dbtx.ReadBucket(wtxmgrNamespaceKey)
+
+		_, tipHeight := w.TxStore.MainChainTip(txmgrNs)
+		ticketAge := int64(tipHeight - details.Ticket.Height())
+
+		// Check if ticket age is not yet mature
+		if ticketAge <= int64(w.ChainParams().TicketMaturity) {
+			ticketStatus = TicketStatusImmature
+			// Check if ticket age is over TicketExpiry limit and therefore expired
+		} else if ticketAge >= int64(w.ChainParams().TicketExpiry) {
+			ticketStatus = TicketStatusExpired
+		} else {
+			// Final check to see if ticket was missed otherwise it's live
+			live, err := w.chainClient.ExistsLiveTicket(&details.Ticket.Hash)
+			if err != nil {
+				log.Errorf("Unable to check if ticket was live for ticket status: %v", &details.Ticket.Hash)
+				return nil
```

Comment:
This behavior is not documented at all and will result in a runtime panic later if it occurs.

Reply:
So you think it should just set the status to TicketUnknown, log the error and carry on?

Reply:
yes